delphi-frontend/src/main/java/au/com/integradev/delphi/type/intrinsic/IntrinsicArgumentMatcher.java

```diff
@@ -41,6 +41,11 @@ public final class IntrinsicArgumentMatcher extends TypeImpl {
           "<ordinal>",
           type -> type.isInteger() || type.isBoolean() || type.isEnum() || type.isChar());
 
+  public static final Type ANY_TYPED_POINTER =
+      new IntrinsicArgumentMatcher(
+          "<typed pointer>",
+          type -> type.isPointer() && ((PointerType) type).dereferencedType().size() != 0);
```

Comment:
Is there a reason you're using `dereferencedType().size() != 0` instead of `dereferencedType().isUntyped()`?

Reply:
I see this would fail in the nil pointer case - in that case, maybe the following?
It is unintuitive to me that a size of 0 means "untyped or nil" - especially since size doesn't really make sense in these scenarios. I still think it would be good to have it be more explicit.

```java
type -> type.isPointer() && !((PointerType) type).dereferencedType().isUntyped() && !((PointerType) type).isNilPointer());
```